lib/mixins/property-effects.js

```diff
@@ -1533,7 +1557,30 @@ export const PropertyEffects = dedupingMixin(superClass => {
      */
     _flushProperties() {
       this.__dataCounter++;
-      super._flushProperties();
+      if (legacyNoBatch && !this.__dataReady) {
+        // The first time this element is being flushed, propagate pending
+        // data down the tree first; this approximates 1.x's distributeConfig
+        this._propagatePropertyChanges(this.__dataPending, {}, false);
+        // Flushing clients causes this element to become __dataReady
+        this._flushClients();
+        // Capture element data and flush properties one-by one to defeat
+        // Polymer 2.x's batched _propertiesChanged scheme; this approximates
+        // 1.x's applyConfig
```

Comment:
Add a note about how  all property getters will return undefined by virtue of resetting `__data`

Reply:
Done